mathics/builtin/manipulate.py

```diff
@@ -0,0 +1,292 @@
+                                   step=di_value)
+            else:
+                widget = FloatSlider(value=idefault_value, min=imin_value, max=imax_value,
+                                     step=di_value)
+            self._add_widget(widget, i.get_name(), lambda x: from_python(x), ilabel)
+
+    def _add_discrete_options_var(self, symbol, options, idefault, ilabel):
+        formatted_options = []
+        for i, option in enumerate(options.leaves):
+            data = self._evaluation.format_all_outputs(option)
+            formatted_options.append((data['text/plain'], i))
+
+        default_index = 0
+        for i, option in enumerate(options.leaves):
+            if option.same(idefault):
+                default_index = i
+
+        widget = ToggleButtons(options=formatted_options, value=default_index)
+        self._add_widget(widget, symbol.get_name(), lambda i: options.leaves[i], ilabel)
+
+    def apply_min_max_var1(self, i, imin, imax):
+        '{i_Symbol, imin_?RealNumberQ, imax_?RealNumberQ}'
+        self._add_min_max_var(i, imin, imax, imin, i)
+
+    def apply_min_max_var2(self, i, imin, imax, idefault):
+        '{{i_Symbol, idefault_?RealNumberQ}, imin_?RealNumberQ, imax_?RealNumberQ}'
+        self._add_min_max_var(i, imin, imax, idefault, i)
+
+    def apply_min_max_var3(self, i, imin, imax, idefault, ilabel):
+        '{{i_Symbol, idefault_?RealNumberQ, ilabel_}, imin_?RealNumberQ, imax_?RealNumberQ}'
+        self._add_min_max_var(i, imin, imax, idefault, ilabel)
+
+    def apply_min_max_step_var1(self, i, imin, imax, di):
+        '{i_Symbol, imin_?RealNumberQ, imax_?RealNumberQ, di_?RealNumberQ}'
+        self._add_min_max_step_var(i, imin, imax, imin, di, i)
+
+    def apply_min_max_step_var2(self, i, imin, imax, di, idefault):
+        '{{i_Symbol, idefault_?RealNumberQ}, imin_?RealNumberQ, imax_?RealNumberQ, di_?RealNumberQ}'
+        self._add_min_max_step_var(i, imin, imax, idefault, di, i)
+
+    def apply_min_max_step_var3(self, i, imin, imax, di, idefault, ilabel):
+        '{{i_Symbol, idefault_?RealNumberQ, ilabel_}, imin_?RealNumberQ, imax_?RealNumberQ, di_?RealNumberQ}'
+        self._add_min_max_step_var(i, imin, imax, idefault, di, ilabel)
+
+    def apply_discrete_options_var1(self, i, options):
+        '{i_Symbol, options_List}'
+        if len(options.leaves) > 0:
+            self._add_discrete_options_var(i, options, options.leaves[0], i)
+
+    def apply_discrete_options_var2(self, i, options, idefault):
+        '{{i_Symbol, idefault_}, options_List}'
+        self._add_discrete_options_var(i, options, idefault, i)
+
+    def apply_discrete_options_var3(self, i, options, idefault, ilabel):
+        '{{i_Symbol, idefault_, ilabel_}, options_List}'
+        self._add_discrete_options_var(i, options, idefault, ilabel)
+
+
+class ListAnimate(Builtin):
+    rules = {
+        'ListAnimate[l_List]': 'Manipulate[Part[l, i], {i, 1, Length[l], 1}]'
+    }
```

Comment:
Might be worth adding a comment here that this is just a workaround, i.e. `ListAnimate` should really generate an animation. I'm wondering if we'd be better to remove this entirely.